python/lsst/daf/butler/core/butlerConfig.py

```diff
@@ -0,0 +1,104 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Configuration classes specific to the Butler
+"""
+
+import os
+
+from .config import Config
+from .utils import doImport
+
+__all__ = ("ButlerConfig",)
+
+
+class ButlerConfig(Config):
+    """Contains the configuration for a `Butler`
+
+    The configuration is read and merged with default configurations for
+    the particular classes. The defaults are read from
+    ``$DAF_BUTLER_DIR/config`` using names specified by the appropriate classes
+    for registry and datastore, and including the standard schema and storage
+    class definitions.
+
+    Parameters
+    ----------
+    other : `str`
+        Path to butler configuration YAML file.
+    """
+
+    def __init__(self, other):
+
+        # Create an empty config for us to populate
+        super().__init__()
+
+        # Find the butler configs
+        self.defaultsDir = None
+        if "DAF_BUTLER_DIR" in os.environ:
+            self.defaultsDir = os.path.join(os.environ["DAF_BUTLER_DIR"], "config")
```

Comment:
I still wonder if having just one level of overrides is future proof enough. As opposed to a `$DAF_BUTLER_CONFIG_PATH`.

Reply:
I can do that. I thought we agreed that once we have the merging code in one place we can easily add it but since I'm there I'll add this other variable as well.

Reply:
Do you envisage the `$DAF_BUTLER_CONFIG_PATH` containing a set of search paths that look identical to the on in `daf_butler/config` (so we look for `storageClasses.yaml`, etc)? And that the defaults should be filled in by reading butler configs first, then the path in reverse order?

Reply:
I think we should hold off on this until we have a use case for it.  I tend to think that having extra paths that aren't connected to either a repo or a software version could do more harm than good.

Reply:
I implemented this last night... How about I push the change this morning as a new commit and you can take a look.